tls/s2n_quic_support.c

```diff
@@ -90,6 +90,24 @@ int s2n_connection_set_secret_callback(struct s2n_connection *conn, s2n_secret_c
     return S2N_SUCCESS;
 }
 
+/* Currently we need an API that quic can call to process post-handshake messages. Ideally
+ * we could re-use the s2n_recv API but that function needs to be refactored to support quic.
+ * For now we just call this API.
+ */
+int s2n_connection_process_post_handshake_message(struct s2n_connection *conn)
+{
+    POSIX_ENSURE_REF(conn);
+
+    uint8_t message_type = 0;
+    POSIX_GUARD_RESULT(s2n_quic_read_handshake_message(conn, &message_type));
+
```

Comment:
Yeah it does use handshake.io for the handshake message header. I'm using it because the header needs to be stored in a stuffer to save the place of the write cursor if the header is fragmented. I'm not sure what other available stuffers I can use here.

Reply:
For non-quic, we solve that problem with https://github.com/aws/s2n-tls/blob/main/tls/s2n_post_handshake.h#L27

Reply:
Minimally, you probably need to leave a comment calling it the potential send/recv concurrency issue caused by handshake.io.

Reply:
Added a comment.